HLTrigger/special/plugins/HLTCTPPSLocalTrackFilter.cc

```diff
@@ -0,0 +1,217 @@
+  stripLocalTrackInputTag_   (iConfig.getParameter< edm::InputTag > ("stripLocalTrackInputTag")),
+  diamondLocalTrackInputTag_ (iConfig.getParameter< edm::InputTag > ("diamondLocalTrackInputTag")),
+  usePixel_                  (iConfig.getParameter< bool          > ("usePixel")),
+  useStrip_                  (iConfig.getParameter< bool          > ("useStrip")),
+  useDiamond_                (iConfig.getParameter< bool          > ("useDiamond")),
+  minTracks_                 (iConfig.getParameter< int           > ("minTracks")),
+  minTracksPerArm_           (iConfig.getParameter< int           > ("minTracksPerArm")),
+  maxTracks_                 (iConfig.getParameter< int           > ("maxTracks")),
+  maxTracksPerArm_           (iConfig.getParameter< int           > ("maxTracksPerArm")),
+  maxTracksPerPot_           (iConfig.getParameter< int           > ("maxTracksPerPot"))
+{
+  if(usePixel_)
+    pixelLocalTrackToken_   = consumes<edm::DetSetVector<CTPPSPixelLocalTrack>>(pixelLocalTrackInputTag_);
+  if(useStrip_)
+    stripLocalTrackToken_   = consumes<edm::DetSetVector<TotemRPLocalTrack>>(stripLocalTrackInputTag_);
+  if(useDiamond_)
+    diamondLocalTrackToken_ = consumes<edm::DetSetVector<CTPPSDiamondLocalTrack>>(diamondLocalTrackInputTag_);
+
+  LogDebug("") << "HLTCTPPSLocalTrackFilter: pixelTag/stripTag/diamondTag/usePixel/useStrip/useDiamond/minTracks/minTracksPerArm/maxTracks/maxTracksPerArm/maxTracksPerPot : "
+               << pixelLocalTrackInputTag_.encode() << " "
+               << stripLocalTrackInputTag_.encode() << " "
+               << diamondLocalTrackInputTag_.encode() << " "
+               << usePixel_ << " "
+               << useStrip_ << " "
+               << useDiamond_ << " "
+               << minTracks_ << " "
+               << minTracksPerArm_ << " "
+               << maxTracks_ << " "
+               << maxTracksPerArm_ << " "
+               << maxTracksPerPot_;
+}
+
+//
+// member functions
+//
+bool HLTCTPPSLocalTrackFilter::filter(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  int arm45Tracks = 0;
+  int arm56Tracks = 0;
+  std::map<uint32_t, int> tracksPerPot;
+
+  typedef edm::Ref<edm::DetSetVector<CTPPSPixelLocalTrack>> PixelRef;
+  typedef edm::Ref<edm::DetSetVector<TotemRPLocalTrack>> StripRef;
+  typedef edm::Ref<edm::DetSetVector<CTPPSDiamondLocalTrack>> DiamondRef;
+
+  //   Note that there is no matching between the tracks from the several roman pots
+  // so tracks from separate pots might correspond to the same particle.
+  // When the pixels are used in more than one RP (in 2018), then the same situation can
+  // happen within the pixels themselves.
+  if(usePixel_) // Pixels correspond to RP 220 in 2017 data
+  {
+    edm::Handle<edm::DetSetVector<CTPPSPixelLocalTrack>> pixelTracks;
+    iEvent.getByToken(pixelLocalTrackToken_, pixelTracks);
+
+    for(const auto &rpv : (*pixelTracks))
+    {
+      CTPPSPixelDetId id(rpv.id);
```

Comment:
const CTPPSPixelDetId& id(rpv.id) to avoid copying to a new menory location

Reply:
The id variable (of the type CTPPSPixelDetId) is constructed at this line from the rpv.id (of the type uint32_t). Enforcing const, could be used, to ensure the variable is not changed, but the reference should not be used because the variable is constructed on this line.
The variable id is constructed because it gives a richer information, in particular allowing to identify which arm of the detector, which is used below.